### PR TITLE
Add more instrumentation

### PR DESCRIPTION
**Pull request by trask**
Adds the following instrumentation:
* opentelemetry-annotations
* reactor-netty (this is important to spring ecosystem)
* kotlinx-coroutintes (this was split out of java-concurrent instrumentation)
* tomcat (this captures server requests earlier, which can lead to correctly capturing response code in cases where it wasn't previously)

Only added smoke test for `opentelemetry-annotations`. The others I think are ok to only rely on OpenTelemetry tests.

### CODE REVIEW

File: test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java [hunk 1]
```diff
@@ -12,13 +13,22 @@ public String root() {
         return "OK";
     }
 
-
-    @GetMapping("/test")
-    public String test() {
+    @GetMapping("/test-api")
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java:16]: nit: for consistency, can we use the regular camel case for @GetMapping?  i searched all the naming for @GetMapping, i don't see this new naming pattern.

File: test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java [hunk 2]
```diff
@@ -14,8 +14,8 @@
 public class OpenTelemetryApiSupportTest extends AiSmokeTest {
 
     @Test
-    @TargetUri("/test")
-    public void doMostBasicTest() throws Exception {
+    @TargetUri("/test-api")
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java:17]: camel case please?

File: test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java [hunk 3]
```diff
@@ -35,5 +35,51 @@ public void doMostBasicTest() throws Exception {
         assertEquals("myspanname", rdd.getName());
 
         assertTrue(rd.getSuccess());
+
+        assertParentChild(rd.getId(), rdEnvelope, rddEnvelope);
+    }
+
+    @Test
+    @TargetUri("/test-annotations")
```
Comment (heyams) [inline test/smoke/testApps/OpenTelemetryApiSupport/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/OpenTelemetryApiSupportTest.java:43]: camel case?

**Review by heyams — APPROVED** [review]
A very minor comment about naming.

File: test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java (hunk 1 above)
Comment (trask) [reply test/smoke/testApps/OpenTelemetryApiSupport/src/main/java/com/microsoft/ajl/simple/TestController.java:16]: hyphens are much more common then camelCase for urls, better I think to change the other places

**Pull request merged**